src/components/Identity/createComponent.js

```diff
@@ -23,17 +25,78 @@ export default (processIdSyncs, config, logger, consent, eventManager) => {
     orgId,
     IDENTITY_COOKIE_KEY
   );
+  let deferredForOptInOld;
```

Comment:
I will review in more details asap but I think this whole logic should be in the `migration` module, and it should only run when migration is turned on.